other/upgrade_2-1_postgresql.sql

```diff
@@ -1170,6 +1170,116 @@ ALTER TABLE {$db_prefix}members
 	DROP notify_announcements;
 ---#
 
+---# Creating alert prefs for watched topics
+---{
+	$_GET['a'] = isset($_GET['a']) ? (int) $_GET['a'] : 0;
+	$step_progress['name'] = 'Creating alert preferences for watched topics';
+	$step_progress['current'] = $_GET['a'];
+
+	$limit = 100000;
+	$is_done = false;
+
+	$request = $smcFunc['db_query']('', 'SELECT COUNT(*) FROM {db_prefix}log_notify WHERE id_member <> 0 AND id_topic <> 0');
+	list($maxTopics) = $smcFunc['db_fetch_row']($request);
+	$smcFunc['db_free_result']($request);
+
+	while (!$is_done)
+	{
+		nextSubStep($substep);
+		$inserts = array();
+
+		$request = $smcFunc['db_query']('', '
+			SELECT id_member, id_topic
+			FROM {db_prefix}log_notify
+			WHERE id_member <> 0 AND id_topic <> 0
+			LIMIT {int:start}, {int:limit}',
+			array(
+				'db_error_skip' => true,
+				'start' => $_GET['a'],
+				'limit' => $limit,
+			)
+		);
+		if ($smcFunc['db_num_rows']($request) != 0)
+		{
+			while ($row = $smcFunc['db_fetch_assoc']($request))
+			{
+				$inserts[] = array($row['id_member'], 'topic_notify_' . $row['id_topic'], 1);
+			}
+			$smcFunc['db_free_result']($request);
```

Comment:
should be db_free_result call everytime undepended of the amount of rows?

Reply:
Do you mean that the call should be outside of the if case?